euphoria-flink/src/main/java/cz/seznam/euphoria/flink/streaming/ReduceStateByKeyTranslator.java

```diff
@@ -137,47 +141,28 @@ public long extractTimestamp(FlinkElement element) {
     }
   }
 
-  private static class WindowAssigner implements MapFunction<FlinkElement, KeyedMultiWindowedElement>,
-          ResultTypeQueryable<KeyedMultiWindowedElement> {
+  private static class WindowAssignerOperator
+          extends AbstractStreamOperator<KeyedMultiWindowedElement>
+          implements OneInputStreamOperator<StreamingElement, KeyedMultiWindowedElement> {
 
-    private final Windowing windowing;
-    private final UnaryFunction keyExtractor;
-    private final UnaryFunction valueExtractor;
-    private final UnaryFunction eventTimeAssigner;
-
-    public WindowAssigner(Windowing windowing,
-                          UnaryFunction keyExtractor,
-                          UnaryFunction valueExtractor,
-                          UnaryFunction eventTimeAssigner) {
-      this.windowing = windowing;
-      this.keyExtractor = keyExtractor;
-      this.valueExtractor = valueExtractor;
-      this.eventTimeAssigner = eventTimeAssigner;
+    private final WindowAssigner windowAssigner;
+
+    private WindowAssignerOperator(WindowAssigner windowAssigner) {
+      this.windowAssigner = windowAssigner;
```

Comment:
I've seen Flink's `StreamMap` (which was previously used for the work done in this new `WindowAssiginerOperator`) uses `chainingStrategy = ChainingStrategy.ALWAYS;` while the default value is `.HEAD`) Can you try that out and see if it has any effect on the executed DAG?